modules/aql/arrow-query-language/src/main/kotlin/arrow/aql/GroupBy.kt

```diff
@@ -10,7 +10,7 @@ interface GroupBy<F> {
 
   fun foldable(): Foldable<F>
 
-  infix fun <A, Z, X> Query<F, A, Z>.groupBy(group: (Z) -> X): Query<ForId, Map<X, List<Z>>, Map<X, List<Z>>> =
+  infix fun <A, Z, X> Query<F, A, Z>.groupBy(group: Z.() -> X): Query<ForId, Map<X, List<Z>>, Map<X, List<Z>>> =
```

Comment:
This is something I thought at some point could be more widespread. I'm not sure about the consequences...